addons/im_livechat/models/discuss_channel.py

```diff
@@ -216,14 +216,49 @@ def _message_post_after_hook(self, message, msg_vals):
         """
         This method is called just before _notify_thread() method which is calling the _to_store()
         method. We need a 'chatbot.message' record before it happens to correctly display the message.
-        It's created only if the mail channel is linked to a chatbot step.
+        It's created only if the mail channel is linked to a chatbot step. We also need to save the
+        user answer if the current step is a question selection.
         """
         if self.chatbot_current_step_id:
-            self.env['chatbot.message'].sudo().create({
-                'mail_message_id': message.id,
-                'discuss_channel_id': self.id,
-                'script_step_id': self.chatbot_current_step_id.id,
-            })
+            selected_answer = (
+                self.env["chatbot.script.answer"]
+                .browse(self.env.context.get("selected_answer_id"))
+                .exists()
+            )
+            if selected_answer in self.chatbot_current_step_id.answer_ids:
+                # sudo - chatbot.message: finding the question message to update the user answer is allowed.
+                question_msg = (
+                    self.env["chatbot.message"]
+                    .sudo()
+                    .search(
+                        [
+                            ("discuss_channel_id", "=", self.id),
+                            ("script_step_id", "=", self.chatbot_current_step_id.id),
+                        ],
+                        order="id DESC",
+                        limit=1,
+                    )
+                )
+                question_msg.user_script_answer_id = selected_answer
+                if store := self.env.context.get("message_post_store"):
+                    store.add(
+                        "ChatbotStep",
+                        {
+                            "id": (question_msg.script_step_id.id, self.id),
+                            "message": question_msg.mail_message_id.id,
+                            "scriptStep": question_msg.script_step_id.id,
+                            "selectedAnswer": selected_answer.id,
+                        },
+                    )
```

Comment:
Ok for `selectedAnswer` to manually put it for now, as this is a fake field for JS only.

For the rest, I'm a bit surprise we have to provide by hand. Couldn't we just use the `message._to_store`? It seems to pass `ChatbotStep` including answer of the message. Since this is a `message_post_after_hook`, I would assume the message was created and it seems to be linked to the right.

At a glance of code, the LOC below `chatbot.message`.create() is required for sending to store. Couldn't we just `store.add()` after these LOCs? Oh yeah there's the fact it needs to be committed... Couldn't we put the bus_send in the post_commit?

Reply:
`question_msg.mail_message_id._to_store()` has apparently all we need, so I just replaced all this by `store.add(question_msg.mail_message_id)`